databricks-sdk-java/src/main/java/com/databricks/sdk/core/oauth/AzureGithubOidcCredentialsProvider.java

```diff
@@ -0,0 +1,100 @@
+ * an Azure Active Directory Federated Identity. It authenticates with Azure by exchanging GitHub's
+ * OIDC token for an Azure Active Directory (AAD) Service Principal OAuth token. This class handles
+ * the process of obtaining, refreshing, and attaching the necessary tokens to each HTTP request.
+ */
+public class AzureGithubOidcCredentialsProvider implements CredentialsProvider {
+  private final ObjectMapper mapper = new ObjectMapper();
+
+  @Override
+  public String authType() {
+    return "github-oidc-azure";
+  }
+
+  @Override
+  public HeaderFactory configure(DatabricksConfig config) {
+    if (!config.isAzure()
+        || config.getAzureClientId() == null
+        || config.getAzureTenantId() == null
+        || config.getHost() == null) {
+      return null;
+    }
+
+    Optional<String> idToken = requestIdToken(config);
+    if (!idToken.isPresent()) {
+      return null;
+    }
+
+    TokenSource tokenSource =
+        new OidcTokenSource(
+            config.getHttpClient(),
+            "",
+            config.getClientId(),
+            config.getEffectiveAzureLoginAppId(),
+            idToken.get(),
+            "urn:ietf:params:oauth:client-assertion-type:jwt-bearer");
+
+    return () -> {
+      Map<String, String> headers = new HashMap<>();
+      headers.put("Authorization", "Bearer " + tokenSource.getToken().getAccessToken());
+      return headers;
+    };
+  }
+
+  /**
+   * Requests an Azure access token using GitHub's OIDC token.
+   *
+   * @param config The DatabricksConfig instance containing the required authentication parameters.
+   * @return An optional Azure access token.
+   */
+  private Optional<String> requestIdToken(DatabricksConfig config) {
+    if (config.getActionsIdTokenRequestUrl() == null
+        || config.getActionsIdTokenRequestToken() == null) {
+      return Optional.empty();
+    }
+
+    String requestUrl =
+        config.getActionsIdTokenRequestUrl() + "&audience=api://AzureADTokenExchange";
+    Request req =
+        new Request("GET", requestUrl)
+            .withHeader("Authorization", "Bearer " + config.getActionsIdTokenRequestToken());
+
+    Response resp;
+    try {
+      resp = config.getHttpClient().execute(req);
+      if (resp.getStatusCode() != 200) {
+        throw new DatabricksException(
+            "Failed to request ID token: status code "
+                + resp.getStatusCode()
+                + ", response body: "
+                + resp.getBody());
+      }
+    } catch (IOException e) {
+      throw new DatabricksException(
+          "Failed to request ID token from " + requestUrl + ":" + e.getMessage(), e);
+    }
+
+    ObjectNode jsonResp;
+    try {
+      jsonResp = mapper.readValue(resp.getBody(), ObjectNode.class);
+    } catch (IOException e) {
+      throw new DatabricksException(
+          "Failed to request ID token: corrupted token: " + e.getMessage());
+    }
+
+    return Optional.ofNullable(jsonResp.get("value").textValue());
+  }
```

Comment:
same here, we could split this into another method to parse once we get the response above.

Reply:
See comment above, I'd tend to keep as is for similar reason but I'm happy to split if you feel strongly against it.